src/main/js/controls/Gantt/GanttConfig.js

```diff
@@ -167,6 +167,46 @@ const getPadding = (config, inputPadding) => {
         };
     }
 };
+/**
+ * Helper function to set the right legend padding values based on input JSON.
+ *
+ * @private
+ * @param {object} config - config which needs to be updated
+ * @param {object} inputLegendPadding - input legend padding provided via input JSON.
+ * @returns {object} - padding for Legend
+ */
+const getLegendPadding = (config, inputLegendPadding) => {
+    // If legendPadding is provided, update the config object with the provided values, else update it with the default constants
+    if (utils.isDefined(config.legendPadding)) {
+        return {
+            top: getDefaultValue(
+                inputLegendPadding.top,
+                constants.LEGEND_PADDING.top
+            ),
+            bottom: getDefaultValue(
+                inputLegendPadding.bottom,
+                constants.LEGEND_PADDING.bottom
+            ),
+            right: getDefaultValue(
+                inputLegendPadding.right,
+                constants.LEGEND_PADDING.right
+            ),
+            left: getDefaultValue(
+                inputLegendPadding.left,
+                constants.LEGEND_PADDING.left
+            ),
+            hasCustomLegendPadding: true
+        };
+    } else {
+        return {
+            top: constants.LEGEND_PADDING.top,
+            bottom: constants.LEGEND_PADDING.bottom,
+            right: constants.LEGEND_PADDING.right,
+            left: constants.LEGEND_PADDING.left,
+            hasCustomLegendPadding: false
+        };
+    }
+};
```

Comment:
if consumer want to provide custom padding info, is it mandatory to provide values all four padding(top, bottom, left and right)?

Reply:
@AshishMotanamGurunadham That's a good question.  

If consumer provider padding `top` and `bottom` values, `left` and `right` properties are taken from our default constant file.

`getDeafultValue` helper method will do that. You can see the implementation of the method [here](https://github.com/cerner/carbon-graphs/blob/105abac166/src/main/js/core/BaseConfig/helper.js#L44)